mathesar/api/db/viewsets/queries.py

```diff
@@ -50,6 +50,35 @@ def columns(self, request, pk=None):
             output_col_desc = query.output_columns_described
             return Response(output_col_desc)
 
+    @action(methods=['get'], detail=True)
+    def results(self, request, pk=None):
+        paginator = TableLimitOffsetPagination()
+        query = self.get_object()
+        if query.not_partial:
```

Comment:
If the query is partial, it will end up with a 500 error because a response is not returned if the condition is `True`.

Reply:
Thanks for the catch on that. I'll make the change there (and in another place with similar logic).